src/Distributed/halo_communication.jl

```diff
@@ -54,6 +54,12 @@ end
 
 const MPIGrid = AbstractGrid{<:Any, <:Any, <:Any, <:Any, <:MultiArch}
 
+function fill_halo_regions!(arrays::NTuple, bcs, grid::MPIGrid, args...; kwargs...)
```

Comment:
Future optimization...

Reply:
yep, for the moment all fields are done individually, but when we introduce buffers we can lump them together in the MPI exchange